custom/icds_reports/views.py

```diff
@@ -2230,30 +2227,43 @@ def get(self, request, *args, **kwargs):
         length = GOVERNANCE_API_HOME_VISIT_RECORDS_PAGINATION
         query_filters = {'aggregation_level': AggregationLevels.AWC,
                          'num_launched_awcs': 1,
-                         'awc_id__gt': last_awc_id
+                         'awc_id__gt': last_awc_id,
+                         'state_id': state_id
                          }
-
         order = ['awc_id']
 
         if step == 'home_visit':
-            query_filters['state_id'] = state_id
             data, count = get_home_visit_data(
                 length,
                 year,
                 month,
                 order,
                 query_filters
             )
-            response_json = {
-                'data': data,
-                'metadata': {
-                    'month': month,
-                    'year': year,
-                    'count': count,
-                    'timestamp': india_now()
-                }
-            }
+        elif step == 'beneficiary':
```

Comment:
We spoke in the past about having each step be a separate view (since they are separate URLs) rather than one view with a giant if statement, are you still planning to make that change. I think it would improve readability a lot.

Reply:
The change itself will not be very complex. But that would increase the number of classes and that looks a bit weird if for every dynamic value of URL param we create a new class.

thoughts, Comments?

Reply:
I am not sure I understand the concern here about having one for every value. I think it is cleaner and easier to read if each of the steps has its own view class, because each one would be smaller and less complex. Shared functionality between them (which i think is pretty minimal) could be pulled into a base class or helper function. There is a very static list of the possible classes here and its still quite small

Reply:
Yeah, it was no concern. I was just imagining to have single class would be better rather than having multiple classes. But I see the code looks more clean withmultiple classes thought with increased lines in the code.